src/content/guides/efk-stack-app-oidc/index.md

```diff
@@ -0,0 +1,365 @@
+```yaml=
+apiVersion: v1
+data:
+  config.yml: LS0tCl9tZXRhOgogIHR5cGU6ICJjb25maWciCiAgY29uZmlnX3ZlcnNpb246IDIKCmNvbmZpZzoKICBkeW5hbWljOgogICAgaHR0cDoKICAgICAgYW5vbnltb3VzX2F1dGhfZW5hYmxlZDogZmFsc2UKICAgICAgeGZmOgogICAgICAgIGVuYWJsZWQ6IGZhbHNlCiAgICAgICAgaW50ZXJuYWxQcm94aWVzOiAiLioiCiAgICAgICAgcmVtb3RlSXBIZWFkZXI6ICJ4LWZvcndhcmRlZC1mb3IiCiAgICBhdXRoYzoKICAgICAgYmFzaWNfaW50ZXJuYWxfYXV0aF9kb21haW46CiAgICAgICAgZGVzY3JpcHRpb246ICJBdXRoZW50aWNhdGUgdmlhIEhUVFAgQmFzaWMgYWdhaW5zdCBpbnRlcm5hbCB1c2VycyBkYXRhYmFzZSIKICAgICAgICBodHRwX2VuYWJsZWQ6IHRydWUKICAgICAgICB0cmFuc3BvcnRfZW5hYmxlZDogdHJ1ZQogICAgICAgIG9yZGVyOiAwCiAgICAgICAgaHR0cF9hdXRoZW50aWNhdG9yOgogICAgICAgICAgdHlwZTogImJhc2ljIgogICAgICAgICAgY2hhbGxlbmdlOiBmYWxzZQogICAgICAgIGF1dGhlbnRpY2F0aW9uX2JhY2tlbmQ6CiAgICAgICAgICB0eXBlOiAiaW50ZXJuYWwiCgogICAgICBvcGVuaWRfYXV0aF9kb21haW46CiAgICAgICAgaHR0cF9lbmFibGVkOiB0cnVlCiAgICAgICAgdHJhbnNwb3J0X2VuYWJsZWQ6IHRydWUKICAgICAgICBvcmRlcjogMQogICAgICAgIGh0dHBfYXV0aGVudGljYXRvcjoKICAgICAgICAgIHR5cGU6ICJvcGVuaWQiCiAgICAgICAgICBjaGFsbGVuZ2U6IGZhbHNlCiAgICAgICAgICBjb25maWc6CiAgICAgICAgICAgIGVuYWJsZV9zc2w6IGZhbHNlCiAgICAgICAgICAgIHZlcmlmeV9ob3N0bmFtZXM6IGZhbHNlCiAgICAgICAgICAgIHJvbGVzX2tleTogZ3JvdXBzCiAgICAgICAgICAgIHN1YmplY3Rfa2V5OiBwcmVmZXJyZWRfdXNlcm5hbWUKICAgICAgICAgICAgb3BlbmlkX2Nvbm5lY3RfdXJsOiBodHRwczovL2xvZ2luLm1pY3Jvc29mdG9ubGluZS5jb20vMzFmNzViZjktM2Q4Yy00NjkxLTk1YzAtODNkZDcxNjEzZGI4L3YyLjAvLndlbGwta25vd24vb3BlbmlkLWNvbmZpZ3VyYXRpb24KICAgICAgICBhdXRoZW50aWNhdGlvbl9iYWNrZW5kOgogICAgICAgICAgdHlwZTogbm9vcAogICAgYXV0aHo6IHt9Cg==
+kind: Secret
+metadata:
+  creationTimestamp: "2020-11-03T16:24:02Z"
+  name: opendistro-security-config
+  namespace: efk-stack-app
+  resourceVersion: "72854"
+  selfLink: /api/v1/namespaces/efk-stack-app/secrets/opendistro-security-config
+  uid: b3152bce-71d4-45a6-9837-9ba19eec5083
+type: Opaque
+```
+
+**Note**
+The parameter `roles_key` in `config.yml` defines what key from the OIDC token is used to identify the users roles on kibana. Here it is setup to use the users group IDs. How to configure azure to send the users group IDs in the OIDC token is covered in the section **Register app on azure portal and configure jwt token**
+
+### Configure internal-users database
+The internal-users secret can be used to add initial users to the opendistro security plugins internal user database.
+
+The file format requires hashed passwords. Use https://bcrypt-generator.com/ or refer to the [opendistro-security guide](https://opendistro.github.io/for-elasticsearch-docs/docs/security/configuration/yaml/#internal_usersyml).
+
+Similar to the previous section, initial users to add to the opendistro-security plugins internal user database are provided through a secret. This secret also need to be present in the target namespace for efk-stack-app to deploy successfully.
+
+**Make sure the admin-password matches with the admin password defined in the values.yaml as configured in "Configure OIDC backend for Kibana"**
```

Comment:
"Configure OIDC backend for Kibana" is at the end of this doc. 

Maybe this should be moved to that section? 

This could lose the reader b/c they haven't done this step yet. So they cannot make sure the admin-password matches with that.

Reply:
Actually, where is the "admin-password" in the internal_users.yml file?

Reply:
Is it admin.hash and logstash.hash?

If yes, which one should match with kibana.password? Is it admin.hash?

Maybe worth clarifying...

Reply:
I implemented these change here https://hackmd.io/@XLn9J6qoS5CgCs97V0K5bw/rJkqDsbKv/edit